examples/addon.node/addon.cpp

```diff
@@ -128,192 +128,227 @@ void whisper_print_segment_callback(struct whisper_context * ctx, struct whisper
 
 void cb_log_disable(enum ggml_log_level, const char *, void *) {}
 
-int run(whisper_params &params, std::vector<std::vector<std::string>> &result) {
+class ProgressWorker : public Napi::AsyncWorker {
+ public:
+  ProgressWorker(Napi::Function& callback, whisper_params params, Napi::Function progress_callback, Napi::Env env)
+      : Napi::AsyncWorker(callback), params(params), env(env) {
+    // Create thread-safe function
+    if (!progress_callback.IsEmpty()) {
+      tsfn = Napi::ThreadSafeFunction::New(
+          env,
+          progress_callback,
+          "Progress Callback",
+          0,
+          1
+      );
+    }
+  }
+
+  ~ProgressWorker() {
+    if (tsfn) {
+      // Make sure to release the thread-safe function on destruction
+      tsfn.Release();
+    }
+  }
+
+  void Execute() override {
+    // Use custom run function with progress callback support
+    run_with_progress(params, result);
+  }
+
+  void OnOK() override {
+    Napi::HandleScope scope(Env());
+    Napi::Object res = Napi::Array::New(Env(), result.size());
+    for (uint64_t i = 0; i < result.size(); ++i) {
+      Napi::Object tmp = Napi::Array::New(Env(), 3);
+      for (uint64_t j = 0; j < 3; ++j) {
+        tmp[j] = Napi::String::New(Env(), result[i][j]);
+      }
+      res[i] = tmp;
+    }
+    Callback().Call({Env().Null(), res});
+  }
+
+  // Progress callback function - using thread-safe function
+  void OnProgress(int progress) {
+    if (tsfn) {
+      // Use thread-safe function to call JavaScript callback
+      auto callback = [progress](Napi::Env env, Napi::Function jsCallback) {
+        jsCallback.Call({Napi::Number::New(env, progress)});
+      };
+      
+      tsfn.BlockingCall(callback);
+    }
+  }
+
+ private:
+  whisper_params params;
+  std::vector<std::vector<std::string>> result;
+  Napi::Env env;
+  Napi::ThreadSafeFunction tsfn;
+
+  // Custom run function with progress callback support
+  int run_with_progress(whisper_params &params, std::vector<std::vector<std::string>> &result) {
     if (params.no_prints) {
-        whisper_log_set(cb_log_disable, NULL);
```

Comment:
I think I would be good to stick with 4 space indentation for C++ code here to be consistent with the rest of the code in this file and the rest of the code base. (I realize Node.js uses 2 space indentation but I think it is better not to mix)